### ISSUE TITLE
Division by zero in Topic::getPageCount()

### ISSUE THREAD

**Issue by MacGyer** [mittwald/typo3_forum#228]
Situation: Upgraded from mm_forum to typo3_forum.

The following error occurs when "Last Post" (topic list) or User details -> Latest Posts -> Last Posts is called:

`
#1: PHP Warning: Division by zero in /var/www/dhg/web/typo3conf/ext/typo3_forum/Classes/Domain/Model/Forum/Topic.php line 308
`

Debugging showed that `$this->settings` is not populated in Getter:
https://github.com/mittwald/typo3_forum/blob/develop/Classes/Domain/Model/Forum/Topic.php#L308

TypoScript has been configured like so:
http://typo3-forum.readthedocs.io/en/master/Configuration/Index.html

What to do to circumvent this error?

Edit: Tranlated post to English

**Comment by MacGyer** [comment]
Just for confirmation: The error above occurs on a fresh Typo3 installation with freshly installed forum extension.

Furthermore `$this->settings` is null in this line too:
https://github.com/mittwald/typo3_forum/blob/develop/Classes/Domain/Model/User/FrontendUser.php#L603

This leads to always showing the default avatar image.

System: Typo3 7.6.21

**Comment by julianwinkel** [comment]
We have the same error here. 
While creating a new post, the magic function `$this->injectSettings()` in the class Mittwald\Typo3Forum\Domain\Model\Forum\Topic is not called and `$this->settings` is null.

![-users-julian-desktop-typo3 20exception htm](https://user-images.githubusercontent.com/11630948/30431832-2623d8be-9960-11e7-931c-e9b021acc87c.png)

**Comment by bi11n3r** [comment]
We experience the same error. `settings` are not loaded in the topics model. It would be great if somebody of the team could have a look?

**Comment by itribe** [comment]
Solution:
1) https://monosnap.com/file/V9XsGLCk495YZRAUK8WhNWjxfqTBnX
2) https://monosnap.com/file/gWiGpy5PM5KbtzPLVdfpaVZ1VzSli1

**Comment by bi11n3r** [comment]
Great - thank you. It would be nice to integrate this fix in the extension since XClassing or changing the source is not a very elegant and permanent solution. Any plans to include this update?

**Comment by kpurrmann** [comment]
Resolved in #216

**Comment by gerathjen** [comment]
After editing the **[BUGFIX] Division by zero in Topic::getPageCount()** I get follow error 
```
Call to a member function getSettings() on null 
Error thrown in file
/var/www/staging/releases/33/packages/typo3_forum/Classes/Domain/Model/Forum/Topic.php in line 302
```
I use Typo3 8.7.19 and Mittwald Forum 1.1.0.

![TYPO3 Exception - Google Chrome_048](https://user-images.githubusercontent.com/5717825/60724784-9d255d00-9f37-11e9-8610-6f7c1d4993f3.jpg)
